### DISCUSSION TITLE
Possibly add some dithering algorithms

### DISCUSSION THREAD

**Discussion by dg1727** [libvips/libvips#1997] [Ideas]
Suggestion:  Add dithering algorithms as given on http://www.tannerhelland.com/4660/dithering-eleven-algorithms-source-code/ (obviously excepting the "false Floyd-Steinberg" one and, as far as I am concerned, the two "ordered dither" ones as well).  

This would help in converting greyscale to 1-bit bitmaps, and as noted at the beginning of the referenced article, also has applications in reducing images to any arbitrary bit depth (someone may find that a compressed image with 64 or 128 or however-many colours has the right storage size or download time for something [s]he needs).  

This proposal is for after the VIPS 8.0 release.

**Comment by jcupitt** [comment]
There's a vips plugin here which does a very simple 4x4 ordered dither:

https://github.com/jcupitt/vips-dither

It's apparently useful for hiding banding on 16bpp displays.

**Comment by jcupitt** [comment]
8.10 will automatically dither low bit depth PNG images. There's a `dither` parameter to prevent this.
